src/discounts/views/SaleList/SaleList.tsx

```diff
@@ -156,6 +156,9 @@ export const SaleList: React.FC<SaleListProps> = ({ params }) => {
 
   const handleSort = createSortHandler(navigate, saleListUrl, params);
   const currencySymbol = maybe(() => shop.defaultCurrency, "USD");
+  const onSettingsOpen = !!channelChoices.length
+    ? { onSettingsOpen: () => openModal("settings") }
+    : {};
```

Comment:
I think it is easier to read:
```suggestion
  const onSettingsOpen = !!channelChoices.length
    ? () => openModal("settings")
    : undefined;
```